dbt/adapters/athena/impl.py

```diff
@@ -452,27 +446,26 @@ def _get_catalog_schemas(self, manifest: Manifest) -> AthenaSchemaSearchMap:
             info_schema_name_map.add(relation)
         return info_schema_name_map
 
-    def _get_data_catalog(self, catalog_name):
-        conn = self.connections.get_thread_connection()
-        client = conn.handle
-        with boto3_client_lock:
-            athena_client = client.session.client("athena", region_name=client.region_name, config=get_boto3_config())
-
-        response = athena_client.get_data_catalog(Name=catalog_name)
-        return response["DataCatalog"]
-
-    def list_relations_without_caching(
-        self,
-        schema_relation: AthenaRelation,
-    ) -> List[BaseRelation]:
-        catalog_id = None
-        if schema_relation.database is not None and schema_relation.database.lower() != "awsdatacatalog":
-            data_catalog = self._get_data_catalog(schema_relation.database.lower())
-            # For non-Glue Data Catalogs, use the original Athena query against INFORMATION_SCHEMA approach
-            if data_catalog["Type"] != "GLUE":
-                return super().list_relations_without_caching(schema_relation)
+    def _get_data_catalog(self, database: str) -> Optional[DataCatalogTypeDef]:
+        if database:
+            conn = self.connections.get_thread_connection()
+            client = conn.handle
+            if database.lower() == "awsdatacatalog":
+                with boto3_client_lock:
+                    sts = client.session.client("sts", region_name=client.region_name, config=get_boto3_config())
+                catalog_id = sts.get_caller_identity()["Account"]
+                return {"Name": database, "Type": "GLUE", "Parameters": {"catalog-id": catalog_id}}
             else:
-                catalog_id = data_catalog["Parameters"]["catalog-id"]
+                with boto3_client_lock:
+                    athena = client.session.client("athena", region_name=client.region_name, config=get_boto3_config())
+                return athena.get_data_catalog(Name=database)["DataCatalog"]
```

Comment:
here you are returning the full dictionary from the client call, that can contains way more fields.
Did you check the dict keys of what is returned and maybe consider to do an extra clean up to make code more concise and readable?
e.g. now I have no clue what `athena.get_data_catalog(Name=database)["DataCatalog"]` contains.

Reply:
that's why I offer to use type annotations. `athena.get_data_catalog` method returns `DataCatalogTypeDef`:
```
class DataCatalogTypeDef(_RequiredDataCatalogTypeDef, _OptionalDataCatalogTypeDef):
    pass

_RequiredDatabaseTypeDef = TypedDict(
    "_RequiredDatabaseTypeDef",
    {
        "Name": str,
    },
)
_OptionalDatabaseTypeDef = TypedDict(
    "_OptionalDatabaseTypeDef",
    {
        "Description": str,
        "Parameters": Dict[str, str],
    },
    total=False,
)
```

Reply:
Type hints helps us to understand that the returned type is consistent, I somehow miss that and it helps.

But still, if `athena.get_data_catalog` starts to returns something different then expected, we can have failures downstream, and as I generally use the guideline: `if you fail, fail fast` do we want to consider to just return here what we need, could be easy as

```
_catalog = athena.get_data_catalog(Name=database)["DataCatalog"]
return {"Name": _catalog["Name"], "Type": _catalog["Type"], "Parameters": {"catalog-id": _catalog["Parameters"]["catalog-id"]}}
```

Pros of this approach:
* strict checks that the response contains what we need
* in case the response change we can adapt easy the returned dict 
Cons:
* uglier code

I have no strong opinion about this, if you don't find my comment relevant at all feel free to leave it as it is.